File: leet_1991_find_middle_index___2waysum.py
class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:
        L, R = 0, 0
        total = sum(nums)
        for i in range(len(nums)):
            R = total - nums[i] - L
            if L == R:
                return i
            L += nums[i]
        return -1

File: leet_1991_find_middle_index___2waysum.cpp
class Solution {
public:
    int findMiddleIndex(vector<int>& nums)
    {
        int     L, R, i, sum;

        L = 0;
        R = 0;
        i = -1;
        sum = accumulate(nums.begin(), nums.end(), 0);
        while (++i < (int) nums.size())
        {
            R = sum - L - nums[i];
            if (L == R)
                return (i);
            L += nums[i];
        }
        return (-1);
    }
};
